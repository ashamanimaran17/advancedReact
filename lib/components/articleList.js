import React, { Component } from "react";
import { Article } from "./article.js";
import PropTypes from "prop-types";
export class ArticleList extends Component {
  render() {
    return (
      <div>
        {Object.values(this.props.articles).map((article) => {
          return (
            <Article
              key={article.id}
              article={article}
              getAuthorInfo={this.props.getAuthorInfo}
            />
          );
        })}
      </div>
    );
  }
}
ArticleList.propTypes = {
  articles: PropTypes.object,
  getAuthorInfo: PropTypes.func,
};
