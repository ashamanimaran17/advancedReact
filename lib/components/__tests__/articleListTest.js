import React from "react";
import { ArticleList } from "../articleList.js";
import renderer from "react-test-renderer";
describe("ArticleList", () => {
  it("should render correctly", () => {
    const authors = { a: {}, b: {} };
    const testProps = {
      articles: {
        123: { id: "123", authorId: "a" },
        456: { id: "456", authorId: "b" },
      },
      getAuthorInfo: function (id) {
        return authors[id];
      },
    };
    const tree = renderer
      .create(
        <ArticleList
          articles={testProps.articles}
          getAuthorInfo={testProps.getAuthorInfo}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
