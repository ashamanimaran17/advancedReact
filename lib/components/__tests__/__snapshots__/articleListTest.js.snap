// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList should render correctly 1`] = `
<div>
  <div>
    title: 
    <br />
    date: 
    <br />
    <a>
      author: \`
       
      \`
       
      <br />
    </a>
    body: 
    123
    <br />
  </div>
  <div>
    title: 
    <br />
    date: 
    <br />
    <a>
      author: \`
       
      \`
       
      <br />
    </a>
    body: 
    456
    <br />
  </div>
</div>
`;
